File: lily/@lily/packages/fab/src/net/bodz/violet/schema/art/ArtifactModel.ts
import IValidControl from "@lily/basic/src/net/bodz/lily/concrete/util/IValidControl";
import ArtifactModelTypeInfo from "@lily/violet/src/net/bodz/violet/schema/art/ArtifactModelTypeInfo";
import FabCost from "@lily/violet/src/net/bodz/violet/schema/art/FabCost";
import _ArtifactModel_stuff from "@lily/violet/src/net/bodz/violet/schema/art/_ArtifactModel_stuff";

export class ArtifactModel extends _ArtifactModel_stuff {

    static _typeInfo: ArtifactModelTypeInfo;
    static get TYPE() {
        if (this._typeInfo == null)
            this._typeInfo = ArtifactModelTypeInfo.INSTANCE;
        return this._typeInfo;
    }

    cost?: FabCost
    modelName?: string
    validControl?: IValidControl

    constructor(o?: any) {
        super(o);
    }
}

export default ArtifactModel;

File: lily/fab-types/src/main/java/net/bodz/violet/schema/art/ArtifactModel.java
package net.bodz.violet.schema.art;

import javax.persistence.Table;

import net.bodz.lily.concrete.util.IValidControl;

@Table(schema = ArtifactModel.SCHEMA_NAME, name = ArtifactModel.TABLE_NAME)
public class ArtifactModel
        extends _ArtifactModel_stuff
        implements
            IValidControl {

    private static final long serialVersionUID = 1L;

    public static final int N_MODEL = 40;

    public interface Props {
        String cost = "cost";
        String priceStrategy = "priceStrategy";
    }

    String modelName;

    final FabCost cost = new FabCost();

    public ArtifactModel() {
    }

    @Override
    public String getModelName() {
        return modelName;
    }

    @Override
    public void setModelName(String modelName) {
        this.modelName = modelName;
    }

    public IValidControl getValidControl() {
        return this;
    }

    public void setValidControl(IValidControl validControl) {
        setValid(validControl.isValid());
        setValidSince(validControl.getValidSince());
        setValidUntil(validControl.getValidUntil());
    }

    public FabCost getCost() {
        return cost;
    }

}
